pil_analyzer/src/pil_analyzer.rs

```diff
@@ -653,12 +653,27 @@ impl PILContext {
                     assert!(right_int <= u32::MAX.into());
                     left.pow(right_int)
                 }
-                BinaryOperator::Mod => (left.to_integer() % right.to_integer()).into(),
-                BinaryOperator::BinaryAnd => (left.to_integer() & right.to_integer()).into(),
-                BinaryOperator::BinaryXor => (left.to_integer() ^ right.to_integer()).into(),
-                BinaryOperator::BinaryOr => (left.to_integer() | right.to_integer()).into(),
-                BinaryOperator::ShiftLeft => (left.to_integer() << right.to_integer()).into(),
-                BinaryOperator::ShiftRight => (left.to_integer() >> right.to_integer()).into(),
+                BinaryOperator::Mod => (left.to_arbitrary_integer() % right.to_arbitrary_integer())
+                    .try_into()
+                    .unwrap(),
+                BinaryOperator::BinaryAnd => (left.to_arbitrary_integer()
+                    & right.to_arbitrary_integer())
+                .try_into()
+                .unwrap(),
+                BinaryOperator::BinaryXor => (left.to_arbitrary_integer()
+                    ^ right.to_arbitrary_integer())
+                .try_into()
+                .unwrap(),
+                BinaryOperator::BinaryOr => (left.to_arbitrary_integer()
+                    | right.to_arbitrary_integer())
+                .try_into()
+                .unwrap(),
```

Comment:
Indentation looks weird

Reply:
fmt 🤷 will see what I can do

Reply:
ah, weird... some items are similar to this but have the "correct" indentation

Reply:
This can actually use fixed width integers